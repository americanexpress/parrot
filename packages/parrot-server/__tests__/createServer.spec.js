/*
 * Copyright (c) 2018 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations under
 * the License.
 */

/* eslint-disable global-require */
import request from 'supertest';
import prettyFormat from 'pretty-format';

describe('createServer', () => {
  beforeEach(() => {
    jest.resetModules().clearAllMocks();
  });

  it('can resolve scenarios file from relative path', () => {
    const createServer = require('../bin/utils/createServer');

    expect(() => createServer('./packages/parrot-server/__fixtures__/scenarios.js')).not.toThrow();
  });

  it('can resolve scenarios file from absolute path', () => {
    const createServer = require('../bin/utils/createServer');

    expect(() => createServer(require.resolve('../__fixtures__/scenarios.js'))).not.toThrow();
  });

  it('uses parrot middleware', async () => {
    const createServer = require('../bin/utils/createServer');

    const app = createServer(require.resolve('../__fixtures__/scenarios.js'));
    const response = await request(app).get('/parrot/scenarios');

    expect(response.status).toBe(200);
    expect(prettyFormat(response.body)).toMatchSnapshot();
  });

  it('returns an express application', () => {
    const mockExpressApp = { use: jest.fn() };
    jest.doMock('express', () => () => mockExpressApp);
    const createServer = require('../bin/utils/createServer');

    const app = createServer(require.resolve('../__fixtures__/scenarios.js'));
    expect(app).toEqual(mockExpressApp);
  });
});
