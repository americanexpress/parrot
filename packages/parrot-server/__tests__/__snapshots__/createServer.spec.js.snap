// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createServer uses parrot middleware 1`] = `
"Array [
  Object {
    \\"mocks\\": Array [
      Object {
        \\"request\\": Object {
          \\"method\\": \\"GET\\",
          \\"path\\": \\"/ship_log\\",
        },
        \\"response\\": Object {
          \\"body\\": Array [
            Object {
              \\"captain\\": \\"Captain Hook\\",
              \\"name\\": \\"The Jolly Roger\\",
            },
          ],
          \\"status\\": 200,
        },
      },
    ],
    \\"name\\": \\"has one ship\\",
  },
  Object {
    \\"mocks\\": Array [
      Object {
        \\"request\\": Object {
          \\"method\\": \\"GET\\",
          \\"path\\": \\"/ship_log\\",
        },
        \\"response\\": Object {
          \\"body\\": Array [
            Object {
              \\"captain\\": \\"Captain Hook\\",
              \\"name\\": \\"The Jolly Roger\\",
            },
            Object {
              \\"captain\\": \\"Jack Sparrow\\",
              \\"name\\": \\"The Black Pearl\\",
            },
            Object {
              \\"captain\\": \\"Davy Jones\\",
              \\"name\\": \\"Flying Dutchman\\",
            },
            Object {
              \\"captain\\": \\"Captain Ron\\",
              \\"name\\": \\"The Wanderer\\",
            },
          ],
          \\"status\\": 200,
        },
      },
    ],
    \\"name\\": \\"has more ships\\",
  },
  Object {
    \\"mocks\\": Array [
      Object {
        \\"request\\": Object {
          \\"method\\": \\"GET\\",
          \\"path\\": \\"/ship_log\\",
        },
        \\"response\\": Object {
          \\"status\\": 500,
        },
      },
    ],
    \\"name\\": \\"has a server error\\",
  },
]"
`;
